src/components/engagement/handlers/set-initial-end-date.handler.ts

```diff
@@ -26,6 +34,9 @@ export class SetInitialEndDate implements IEventHandler<SubscribedEvent> {
 
     const engagement = 'engagement' in event ? event.engagement : event.updated;
 
+    if (event instanceof EngagementCreatedEvent && engagement.changeset) {
+      return;
+    }
```

Comment:
I don't think there's a need to prevent this from happening in the changeset...It is just a property change like anything a user would directly change.